plugins/module_utils/implementations/mysql/hash.py

```diff
@@ -0,0 +1,105 @@
+"""Generate MySQL sha256 compatible plugins hash for a given password and salt."""
+
```

Comment:
sources needed.

* https://www.akkadia.org/drepper/SHA-crypt.txt
* https://crypto.stackexchange.com/questions/77427/whats-the-algorithm-behind-mysqls-sha256-password-hashing-scheme/111174#111174
* https://github.com/hashcat/hashcat/blob/master/tools/test_modules/m07400.pm